### PR TITLE
Shrink brand before wrap

### PR DESCRIPTION
**Pull request by AlistairGempf**
Resolves #NUMBER

**Overall change:** Shrinks the brand svg before the script link wraps onto a new line (except on IE which an IE bug has prevented from working). Previously, the script link would wrap onto a new line before the brand svg started shrinking. This PR makes the brand svg shrink as much as possible before putting the script link onto a new line.

**Code changes:**

- Use flex-basis of the svg minimum width
- Use flex-shrink and flex-grow to allow the svg to change width
- Keep the -ms-flex-preferred-size at the max width to prevent [this bug](https://github.com/philipwalton/flexbugs#flexbug-17)

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by AlistairGempf** [comment]
Blocked on #2887

**Comment by AlistairGempf** [comment]
Chromatic is showing 
image on IE. Not sure why it would do that or how to investigate on IE.

**Comment by AlistairGempf** [comment]
The problem is that IE11 calculates the flex spacing based on the element's flex-basis size, then adds that much spacing between the elements, even if the element is larger than its flex-basis.

In order to have the shrink happen before the wrap, the flex basis needs to be the min-width. This leads IE11 to add the spacing required for the min-width even when the svg is at max-width, pushing the script link off the edge of the window.

**Comment by AlistairGempf** [comment]
One possible solution is to use the css `-ms-flex-preferred-size: ${maxWidth / 16}rem` to override the `flex-basis` set in the `flex` property only on IE browsers.

This would prevent the shrinking happening before the wrap, but IE must almost entirely be used on desktop so this loss of functionality will not impact many people and it would display the script link reasonably at all widths.

**Review by hotinglok — APPROVED** [review]
LGTM, tested on storybook and IE on browserstack and the functionality seems to be working fine, nice work!

File: packages/components/psammead-brand/src/index.jsx [hunk 1]
```diff
@@ -71,6 +71,8 @@ const brandWidth = (minWidth, maxWidth) => `
   width: 100%;
   max-width: ${maxWidth / 16}rem;
   min-width: ${minWidth / 16}rem;
+  flex: 1 1 ${minWidth / 16}rem;
+  -ms-flex-preferred-size: ${maxWidth / 16}rem;
```
Comment (sareh) [inline packages/components/psammead-brand/src/index.jsx:75]: The update to `flex` works well on supported browsers. 
For `-ms-flex-preferred-size`, I looked on Browserstack with IE11 and removed this style and did not see any change in behaviour. Am I missing something here?

Comment (AlistairGempf) [reply packages/components/psammead-brand/src/index.jsx:75]: image
image

This is what I see with `-ms-flex-preferred-size` turned off on IE11 - the script switch is off the edge of the screen.

Comment (sareh) [reply packages/components/psammead-brand/src/index.jsx:75]: Seems like I had an issue with the changes in the browserstack dev tools not applying correctly. When I made changes locally I see the same as your screenshot.  
So, I'm happy with the change made!

**Review by sareh — APPROVED** [review]
👍 Great work. I've reviewed in on latest Chrome & Firefox as well as IE11.

**Comment by paruchurisilpa** [comment]
Looks good to me..

**Comment by tochwill** [comment]
@paruchurisilpa is this ok to merge then?

**Pull request merged**